test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java

```diff
@@ -728,19 +743,48 @@ private void disableIRVerificationIfNotFeasible() {
      * For scenarios: Run the tests with the scenario settings and collect all exceptions to be able to run all
      * scenarios without prematurely throwing an exception. Format violations, however, are wrong for all scenarios
      * and thus is reported immediately on the first scenario execution.
+     * @param parallel Run tests concurrently
      */
-    private void startWithScenarios() {
+    private void startWithScenarios(boolean parallel) {
         Map<Scenario, Exception> exceptionMap = new TreeMap<>(Comparator.comparingInt(Scenario::getIndex));
-        for (Scenario scenario : scenarios) {
+        record Outcome(Scenario scenario, TestFormatException tfe, Exception other) {}
+        final Object printLock = new Object();
+
+        Stream<Scenario> stream = parallel ? scenarios.parallelStream() : scenarios.stream();
+        List<Outcome> outcomes = stream.map(scenario -> {
+            Outcome outcome;
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            PrintStream ps = new PrintStream(baos);
             try {
-                start(scenario);
+                start(scenario, ps);
+                outcome = new Outcome(scenario, null, null);
             } catch (TestFormatException e) {
-                // Test format violation is wrong for all the scenarios. Only report once.
-                throw e;
+                outcome = new Outcome(scenario, e, null);
             } catch (Exception e) {
-                exceptionMap.put(scenario, e);
+                outcome = new Outcome(scenario, null, e);
+            } finally {
+                // Print the output to stdout in one go
+                synchronized (printLock) {
+                    String output = baos.toString();
+                    if (!output.isEmpty()) {
+                        System.out.println(output);
+                    }
+                }
```

Comment:
This will probably also not be sorted by scenario index? Could we also just gather it here and then dump it after the stream? Maybe we can put `output` into `Outcome` as well as the exceptions by using a `ConcurrentSkipListMap<Scenario, Outcome>` map.